guides/source/working_with_javascript_in_rails.md

```diff
@@ -3,561 +3,212 @@
+
+  respond_to do |format|
+    if @post.save
+      format.turbo_stream
+    else
+      format.html { render :new, status: :unprocessable_entity }
+    end
+  end
+end
+```
 
-### A Simple Example
+Rails will automatically look for a `.turbo_stream.erb` view file and render that view when found.
 
-Imagine you have a series of users that you would like to display and provide a
-form on that same page to create a new user. The index action of your
-controller looks like this:
+Turbo Stream responses can also be rendered inline in the controller action:
 
 ```ruby
-class UsersController < ApplicationController
-  def index
-    @users = User.all
-    @user = User.new
+def create
+  @post = Post.new(post_params)
+
+  respond_to do |format|
+    if @post.save
+      format.turbo_stream { render turbo_stream: turbo_stream.prepend('posts', partial: 'post') }
+    else
+      format.html { render :new, status: :unprocessable_entity }
+    end
   end
-  # ...
+end
 ```
 
-The index view (`app/views/users/index.html.erb`) contains:
-
-```erb
-<b>Users</b>
-
-<ul id="users">
-<%= render @users %>
-</ul>
+Finally, Turbo Streams can be initiated from a model or a background job using built-in helpers. These broadcasts can be used to update content via a WebSocket connection to all users, keeping page content fresh and bringing your application to life.
 
-<br>
+To broadcast a Turbo Stream from a model combine a model callback like this:
 
-<%= form_with model: @user do |form| %>
-  <%= form.label :name %><br>
-  <%= form.text_field :name %>
-  <%= form.submit %>
-<% end %>
+```ruby
+class Post < ApplicationRecord
+  after_create_commit { broadcast_append_to('posts') }
+end
 ```
 
-The `app/views/users/_user.html.erb` partial contains the following:
+With a WebSocket connection set up on the page that should receive the updates like this:
 
 ```erb
-<li><%= user.name %></li>
+<%= turbo_stream_from "posts" %>
 ```
 
-The top portion of the index page displays the users. The bottom portion
-provides a form to create a new user.
-
-The bottom form will call the `create` action on the `UsersController`. Because
-the form's remote option is set to true, the request will be posted to the
-`UsersController` as an Ajax request, looking for JavaScript. In order to
-serve that request, the `create` action of your controller would look like
-this:
-
-```ruby
-  # app/controllers/users_controller.rb
-  # ......
-  def create
-    @user = User.new(params[:user])
-
-    respond_to do |format|
-      if @user.save
-        format.html { redirect_to @user, notice: 'User was successfully created.' }
-        format.js
-        format.json { render json: @user, status: :created, location: @user }
-      else
-        format.html { render action: "new" }
-        format.json { render json: @user.errors, status: :unprocessable_entity }
-      end
-    end
-  end
-```
+Learn more about Turbo Streams in the [Turbo Handbook](https://turbo.hotwired.dev/handbook/streams).
```

Comment:
```suggestion
```
As with the previous style to linking to external resources, I think linking on the first mention of the linkable thing is more helpful, so let's stick with that.